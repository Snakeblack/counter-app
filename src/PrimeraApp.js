// FC - Funtional Components

import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ( { saludo } ) => {

    // if ( !saludo ){
    //     throw new Error("El saludo es necesario");
    // }

    return (
        <>
            <h1>{ saludo }</h1>
            {/* <pre> { JSON.stringify( saludo, null, 3 ) } </pre> */}
            <p>Texto de prueba</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}


export default PrimeraApp; 