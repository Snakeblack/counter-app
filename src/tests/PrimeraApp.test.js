import React from "react";
import { shallow } from 'enzyme';
// import { render } from "@testing-library/react";

import PrimeraApp from "../PrimeraApp";

describe('Pruebas en <PrimeraApp />', () => {

    // test('Debe de mostrar el mensaje "Bienvenido" ', () => {    
    //     const titulo = 'Bienvenidos';
    //     const { getByText } = render( <PrimeraApp titulo={ titulo } /> );
    //     expect( getByText( titulo ) ).toBeInTheDocument();
    // });
    
    test('Debe demostrar <PrimeraApp /> correctamente', () => {
        
        const titulo = 'Bienvenido';
        const wrapper = shallow( <PrimeraApp titulo= { titulo }/> );

        expect( wrapper ).toMatchSnapshot();

    })
    
    test('Debe de mostrar el subtitulo enviado por props', () => {
        
        const titulo = 'Bienvenido';
        const subTitulo = 'Soy un subtitulo';

        const wrapper = shallow(
            <PrimeraApp
                titulo= { titulo }
                subtitulo={ subTitulo }
            /> 
        );
        
        const textoParrafo = wrapper.find('p').text();
            
        expect( textoParrafo ).toBe( subTitulo );

    })
    

});