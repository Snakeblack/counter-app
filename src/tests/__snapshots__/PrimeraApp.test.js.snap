// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe demostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Bienvenido
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
